tests/rptest/redpanda_cloud_tests/omb_validation_test.py

```diff
@@ -0,0 +1,350 @@
+from rptest.services.producer_swarm import ProducerSwarm
+from rptest.clients.rpk import RpkTool
+from rptest.services.redpanda_cloud import AdvertisedTierConfigs, CloudTierName
+from rptest.services.redpanda import (SISettings, RedpandaServiceCloud)
+from rptest.services.openmessaging_benchmark import OpenMessagingBenchmark
+from rptest.services.openmessaging_benchmark_configs import \
+    OMBSampleConfigurations
+
+KiB = 1024
+MiB = KiB * KiB
+GiB = KiB * MiB
+minutes = 60
+hours = 60 * minutes
+
+
+def get_globals_value(globals, key_name, default=None):
+    _config = {}
+    if RedpandaServiceCloud.GLOBAL_CLOUD_CLUSTER_CONFIG in globals:
+        # Load needed config values from cloud section
+        # of globals prior to actual cluster creation
+        _config = globals[RedpandaServiceCloud.GLOBAL_CLOUD_CLUSTER_CONFIG]
+    return _config.get(key_name, default)
+
+
+class OMBValidationTest(RedpandaTest):
+    def __init__(self, test_ctx: TestContext, *args, **kwargs):
+        self._ctx = test_ctx
+        # Get tier value
+        cloud_tier_str = get_globals_value(self._ctx.globals,
+                                           "config_profile_name",
+                                           default="tier-1-aws")
+        cloud_tier = CloudTierName(cloud_tier_str)
+        extra_rp_conf = None
+        num_brokers = None
+
+        if cloud_tier == CloudTierName.DOCKER:
+            # TODO: Bake the docker config into a higher layer that will
+            # automatically load these settings upon call to make_rp_service
+            config = AdvertisedTierConfigs[CloudTierName.DOCKER]
+            num_brokers = config.num_brokers
+            extra_rp_conf = {
+                'log_segment_size': config.segment_size,
+                'cloud_storage_cache_size': config.cloud_cache_size,
+                'kafka_connections_max': config.connections_limit,
+            }
+
+        super(OMBValidationTest,
+              self).__init__(test_ctx,
+                             node_prealloc_count=11,
+                             *args,
+                             num_brokers=num_brokers,
+                             extra_rp_conf=extra_rp_conf,
+                             cloud_tier=cloud_tier,
+                             disable_cloud_storage_diagnostics=True,
+                             **kwargs)
+
+        self.tier_config = self.redpanda.advertised_tier_config
+        if cloud_tier == CloudTierName.DOCKER:
+            si_settings = SISettings(
+                test_ctx,
+                log_segment_size=self.small_segment_size,
+                cloud_storage_cache_size=self.tier_config.cloud_cache_size,
+            )
+            self.redpanda.set_si_settings(si_settings)
+            self.s3_port = si_settings.cloud_storage_api_endpoint_port
+
+        test_ctx.logger.info(f"Cloud tier {cloud_tier}: {self.tier_config}")
+
+        self.rpk = RpkTool(self.redpanda)
+
+    def _optimal_partition_count(self) -> int:
```

Comment:
How is this used? Is the optimal partition count 1 per shard? I feel this often doesn't actually end up with 1 user partition per shard because of the presence of existing partitions.

E.g., with three shards the existing allocation per might be 1 0 0, so you add a topic with 3 partitions (1 per shard) the result could be say 1 2 1 so the new partitions went 0 2 1, totally uneven.

I guess I should talk about replicas not partitions but the same kind of problem applies.

Reply:
Fair enough, the question then is what is the optimal partition count per shard? OMB will remove any user created topics before running. So only internal RP partitions should remain. Using something like 10 partitions per shard would reduce the unevenness to only a few percent in that case. Of course using more than 10 would get us closer to perfectly uniform as well.

Reply:
I've increased it to 10x for now.

Reply:
> Fair enough, the question then is what is the optimal partition count per shard?

Very good question to which I do not know the answer. I think 10 is fine for now though. More will reduce this potential "error" even more but at some point more is slower too. Possibly 10 is already even slower I guess it depends on the 10. 

10 seems like a semi-realistic number though and avoids leaving any shards doing nothing, etc.